pyomo/contrib/pynumero/cmake/sparse_utils/src/SparseUtils.cpp

```diff
@@ -257,6 +258,58 @@ void sym_csc_diagonal(int* col_pointer,
     }
 }
 
+int sym_csr_allnnz(int* row_pointer,
+                   int* col_indices,
+                   int nrows)
+{
+    int row, col;
+    int off_diagonals = 0;
+    int on_diagonals = 0;
+
+    for (int i=0; i<nrows; ++i)
+    {
+        row = i;
+        for(int jj = row_pointer[row]; jj < row_pointer[row+1]; ++jj)
+        {
+            col = col_indices[jj];
+            if(col==row){
+                ++on_diagonals;
+            }
+            else{
+                ++off_diagonals;
```

Comment:
As written, this method has a computational complexity of O(NNZ).

Not a huge problem, but it might be more performant to only count the diagonals and then subtract to get the off-diagonals.  That would allow you to break out if the inner loop as soon as you see a on-diagonal entry.

I forget: are the columns sorted? If so, you could leverage that and make this more efficient...

Reply:
Removed SparseUtils from `pynumero.cmake` because is no longer used.